Flaky: TestGameServerResourceValidation

<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!

If the matter is security related, please disclose it privately via 
-->


**What happened**:

Occasional failure on e2e tests

```
time="2023-09-06 19:33:45.261" level=info msg="RoleBinding 1694028825/agones-sdk-access is created"
=== RUN   TestGameServerResourceValidation
=== PAUSE TestGameServerResourceValidation
=== CONT  TestGameServerResourceValidation
    gameserver_test.go:953: 
        	Error Trace:	/go/src/agones.dev/agones/test/e2e/gameserver_test.go:953
        	Error:      	Not equal: 
        	            	expected: "spec.template.spec.containers[0].resources.requests[cpu]"
        	            	actual  : "spec.template.spec.containers[0].resources.requests"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-spec.template.spec.containers[0].resources.requests[cpu]
        	            	+spec.template.spec.containers[0].resources.requests
        	Test:       	TestGameServerResourceValidation
--- FAIL: TestGameServerResourceValidation (0.17s)
```

As far as I can tell it's doesn't seem tied to a Kubernetes version (but I could be wrong) - most recently seen on 1.27 standard cluster.

**What you expected to happen**:

To get a consistent error message back. Very odd.

**How to reproduce it (as minimally and precisely as possible)**:

Run a few tests. What's even weirder is on subsequent builds it doesn't fail, but it consistently fails on e2e test retries.

**Anything else we need to know?**:

**Environment**:
- Agones version:  dev
- Kubernetes version (use `kubectl version`): varies
- Cloud provider or hardware configuration: GKE
- Install method (yaml/helm): helm
- Troubleshooting guide log(s): 
  * https://console.cloud.google.com/cloud-build/builds/4e05fa57-2b0b-4bc4-8f43-42b08ad8d242?project=258182270954
- Others: N/A

--- a/test/e2e/gameserver_test.go
+++ b/test/e2e/gameserver_test.go
@@ -21,6 +21,7 @@ import (
 	"net"
 	"os"
 	"os/exec"
+	"sort"
 	"strings"
 	"sync/atomic"
 	"testing"
@@ -925,7 +926,6 @@ func TestGameServerWithoutPort(t *testing.T) {
 func TestGameServerResourceValidation(t *testing.T) {
 	t.Parallel()
 	ctx := context.Background()
-	logger := e2eframework.TestLogger(t)
 	gs := framework.DefaultGameServer(framework.Namespace)
 	mi64 := resource.MustParse("64Mi")
 	gs.Spec.Template.Spec.Containers[0].Resources.Limits[corev1.ResourceMemory] = mi64
@@ -949,8 +949,10 @@ func TestGameServerResourceValidation(t *testing.T) {
 	assert.NotNil(t, err)
 	statusErr, ok = err.(*k8serrors.StatusError)
 	assert.True(t, ok)
-	logger.Debug("TestGameServerResourceValidation gsClient.Create with negative resource status errors: ", statusErr)
 	assert.Len(t, statusErr.Status().Details.Causes, 2)
+	sort.Slice(statusErr.Status().Details.Causes, func(i, j int) bool {
+		return statusErr.Status().Details.Causes[i].Field > statusErr.Status().Details.Causes[j].Field
+	})
 	assert.Equal(t, metav1.CauseTypeFieldValueInvalid, statusErr.Status().Details.Causes[0].Type)
 	assert.Equal(t, "spec.template.spec.containers[0].resources.requests[cpu]", statusErr.Status().Details.Causes[0].Field)
 

